test/Dapr.Actors.Test/ApiTokenTests.cs

```diff
@@ -0,0 +1,163 @@
+            var task = proxy.SetCountAsync(1, new CancellationToken());
+
+            handler.Requests.TryDequeue(out var entry).Should().BeTrue();
+            var headerValues = entry.Request.Headers.GetValues("dapr-api-token");
+            headerValues.Should().Contain("test_token");
+        }
+
+        [Fact]
+        public void CreateProxyWithRemoting_WithApiTokenEnvVarSet()
+        {
+            var actorId = new ActorId("abc");
+            var handler = new TestHttpClientHandler();
+            Environment.SetEnvironmentVariable(Constants.DaprApiTokenEnvironmentVariable, "test_token");
+            var factory = new ActorProxyFactory(null, handler);
+            // Verify that if no api token is set but env var is set, then the default
+            // actor proxy options have the token set
+            Assert.Equal("test_token", factory.DefaultOptions.DaprApiToken);
+            var proxy = factory.CreateActorProxy<ITestActor>(actorId, "TestActor");
+            var task = proxy.SetCountAsync(1, new CancellationToken());
+
+            handler.Requests.TryDequeue(out var entry).Should().BeTrue();
+            var headerValues = entry.Request.Headers.GetValues("dapr-api-token");
+            headerValues.Should().Contain("test_token");
+        }
+
+        [Fact]
+        public void CreateProxyWithRemoting_WithNoApiToken()
+        {
+            var actorId = new ActorId("abc");
+            var handler = new TestHttpClientHandler();
+            var factory = new ActorProxyFactory(null, handler);
+            var proxy = factory.CreateActorProxy<ITestActor>(actorId, "TestActor");
+            var task = proxy.SetCountAsync(1, new CancellationToken());
+
+            handler.Requests.TryDequeue(out var entry).Should().BeTrue();
+            Action action = () => entry.Request.Headers.GetValues("dapr-api-token");
+            action.Should().Throw<InvalidOperationException>();
+        }
+
+        [Fact]
+        public void CreateProxyWithNoRemoting_WithApiToken()
+        {
+            var actorId = new ActorId("abc");
+            var handler = new TestHttpClientHandler();
+            var options = new ActorProxyOptions
+            {
+                DaprApiToken = "test_token",
+            };
+            var factory = new ActorProxyFactory(options, handler);
+            var proxy = factory.Create(actorId, "TestActor");
+            var task = proxy.InvokeMethodAsync("SetCountAsync", 1, new CancellationToken());
+
+            handler.Requests.TryDequeue(out var entry).Should().BeTrue();
+            var headerValues = entry.Request.Headers.GetValues("dapr-api-token");
+            headerValues.Should().Contain("test_token");
+        }
+
+        [Fact]
+        public void CreateProxyWithNoRemoting_WithApiTokenEnvVarSet()
+        {
+            Environment.SetEnvironmentVariable(Constants.DaprApiTokenEnvironmentVariable, "test_token");
```

Comment:
I'll leave it up to you to decide, but I think doing things like this in tests makes the wrong tradeoff, and we're better off not having a test for it. Think about the implications of running tests in parallel and you'll see what I mean, it doesn't scale.

My general advice for testing the parts of the same that interface with the OS (like env-vars):

- Isolate code that touches OS features from the important logic. In a case this, it's simple things like reading the env-var in a property initializer or in the constructor. (already done)
- Don't write tests for constructors and property initializers because it's low value 
- When possible rely on e2e tests - I know we don't have them for this scenario
- Trust yourself that you did a good job with the above and it will all be ok